cmd/maya-apiserver/cstor-operator/spc/handler.go

```diff
@@ -103,7 +105,22 @@ func (c *Controller) enqueueSpc(spc interface{}) {
 
 // synSpc is the function which tries to converge to a desired state for the spc.
 func (c *Controller) syncSpc(spc *apis.StoragePoolClaim) error {
-	err := validate(spc)
+
+	if !spc.DeletionTimestamp.IsZero() {
+		err := handleSPCDeletion(spc)
+		if err != nil {
+			glog.Errorf("failed to sync spc:%s", err.Error())
+		}
+		return nil
+	}
+	err := spcv1alpha1.BuilderForAPIObject(spc).Spc.AddFinalizer(spcv1alpha1.SPCFinalizer)
```

Comment:
IMO if we modified the object we need to proceed with updated object. Instead of old one

Reply:
this seems to be valid comment

Reply:
Yes. Good catch